### PR TITLE
week4 review

### PR DESCRIPTION
**Pull request by Yunhao-Luo**
@ykharitonova

- not sure if Grade updates and Syllabus updates are still necessary

### CODE REVIEW

File: _announcements/week-04.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+- During class and lab, we discussed different data structures such as:
+    - Conditionals
+    - Booleans and Boolean Operators: Comparison Operators and Logical Operators
+    - Branching using the `if`-`elif`-`else` statements
+    - `assert` statements for writing tests of return values
+        
+* The slides and class recordings as well as the videos from the class and lab sections are posted on Gauchospace (look for the ⏩ icon; use your <span style="color:red">**UCSB address**</span> - we won't be granting access to personal gmail accounts.)
+
+* The last section of every chapter in zyBook contains our class notes, which we update afterward with whatever was discussed in class. Sometimes, like this week, there is additional code for you to practice with (the number-guessing game using the `random` module).
```
Comment (ykharitonova) [inline _announcements/week-04.md]: @Yunhao-Luo we don't have it this quarter

File: _announcements/week-04.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+## ⏰ Week 5 - Due dates and Deadlines ⏰
+
+Check the [Calendar]({{site.url}}/{{site.baseurl}}/calendar#week-4), for the due dates. They are the same as they've been before.
+
+Upcoming deadlines are:
+* ⏰  **Tuesday, 2/07 11:59PM**: **LA04**
+* ⏰  **Wednesday, 2/08 11:59PM**: **LA05 CHECKPOINT** Week 5 zyLabs with the word “Checkpoint” in their title
+* ⏰ **Thursday, 2/09**: in-class Quiz 4 on Gauchospace, covering Week 1-4 content.
```
Comment (ykharitonova) [inline _announcements/week-04.md]: No in-class quiz this week to ensure we have enough time to review using IDLE, answer questions and practice using Gradescope.

File: _announcements/week-04.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+> <span style="color:orange"> Keep in mind that the College of Engineering [does not allow this course to be switched to P/NP later]({{site.url}}/{{site.baseurl}}/about#note-about-pnp).</span>
+
+
+## ➡️    Staying on Track
+
+* Ask for clarifications during the [office hours]({{site.url}}/{{site.baseurl}}/schedule/), lab sessions, or post your questions on [Piazza]({{site.aux_links.Piazza}})
+
+🔮 [Looking ahead to Week 5]({{site.url}}/{{site.baseurl}}/calendar#week-5), your tasks for the upcoming days are:
+* Carefully read the instructions! Do NOT rush through them. 💎
+    - Practice writing reusable and generalizable code using functions.
```
Comment (ykharitonova) [inline _announcements/week-04.md]: Take non-function code from zyBooks and see if you can turn it into a function. 
Write tests for it.
Feel free to ask us during the office hours if you are on the right track.

File: _announcements/week-04.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    - Carefully read about how to write the `assert` statements. Notice the general form of the statement: the keyword `assert` followed by a condition (usually with the `==` operator).
```
Comment (ykharitonova) [inline _announcements/week-04.md]: The condition should involve checking the result of the function call. Please review the examples from class and practice writing them on your own. We'll be asking you to write assert statements on the future quizzes.

File: _announcements/week-04.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+If you ever have any questions, please, _first_ consult the [course syllabus]({{site.url}}/{{site.baseurl}}/about) and the [Frequently Asked Questions (FAQ) section]({{site.url}}/{{site.baseurl}}/faq). A lot of [course structure, helpful hints, and expectations]({{site.url}}/{{site.baseurl}}/faq/#weekly-pattern-and-planning-your-work) are also outlined for you on that page.
+
+## Grade updates
+
+_Your zyBook grades have to be manually transferred to Gauchospace, so the synchronization is not instant._
+* Your grades for each week’s activities will be all updated on **Fridays at 10am**.
```
Comment (ykharitonova) [inline _announcements/week-04.md]: @ajitJJadhav When are the grades expected to be updated?

@Yunhao-Luo I had an estimated time for that in the previous week's review, can you please pull it in?

File: _announcements/week-04.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+_Your zyBook grades have to be manually transferred to Gauchospace, so the synchronization is not instant._
+* Your grades for each week’s activities will be all updated on **Fridays at 10am**. 
+* The extensions will be processed as a batch at the end of each week on **Saturdays by 10 am** - you will receive a confirmation when your extension was processed and the grade was updated.
+* Remember that we will not accept retroactive extension requests for previous weeks.
+
+Check the Syllabus about our policies: note that we do not provide additional extensions without proper documentation that shows 1) your name, 2) dates, and 3) the reason/explanation for the extension.
+
+
+## Syllabus updates
```
Comment (ykharitonova) [inline _announcements/week-04.md]: remove this section

File: _announcements/week-04.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+## ➡️    Staying on Track
+
+* Ask for clarifications during the [office hours]({{site.url}}/{{site.baseurl}}/schedule/), lab sessions, or post your questions on [Piazza]({{site.aux_links.Piazza}})
+
+🔮 [Looking ahead to Week 5]({{site.url}}/{{site.baseurl}}/calendar#week-5), your tasks for the upcoming days are:
+* Carefully read the instructions! Do NOT rush through them. 💎
+    - Practice writing reusable and generalizable code using functions.
+    - Carefully read about how to write the `assert` statements. Notice the general form of the statement: the keyword `assert` followed by a condition (usually with the `==` operator).
+
+If you ever have any questions, please, _first_ consult the [course syllabus]({{site.url}}/{{site.baseurl}}/about) and the [Frequently Asked Questions (FAQ) section]({{site.url}}/{{site.baseurl}}/faq). A lot of [course structure, helpful hints, and expectations]({{site.url}}/{{site.baseurl}}/faq/#weekly-pattern-and-planning-your-work) are also outlined for you on that page.
```
Comment (ykharitonova) [inline _announcements/week-04.md]: remove this part

**Review by ykharitonova — CHANGES_REQUESTED** [review]

**Comment by Yunhao-Luo** [comment]
@ykharitonova updated based on requested changes.

**Pull request merged**